modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java

```diff
@@ -277,8 +316,8 @@ public OracleClient(Environment env) {
     responseTimer = env.getSharedResources().getMetricRegistry().timer(env.getMeticNames().getOracleClientGetStamps());
     stampsHistogram = env.getSharedResources().getMetricRegistry().histogram(env.getMeticNames().getOrcaleClientStamps());
 
-    // TODO make thread exit if idle for a bit, and start one when request arrives
-    Thread thread = new Thread(new TimestampRetriever());
+    timestampRetriever = new TimestampRetriever();
+    thread = new Thread(new TimestampRetriever());
```

Comment:
I fixed this e9eb07f .   The test are running, but I Am seeing more exceptions now.  I will make a comment about the line thats causing this.  Not sure what we can do about these extra exceptions.